### PR TITLE
add BRANCH, REPO, TIMEOUT parameters

### PR DESCRIPTION
**Pull request by Josh-Diamond**
https://github.com/rancher/qa-tasks/issues/1056

Small enhancement to allow BRANCH, REPO, and TIMEOUT parameters

### CODE REVIEW

File: tfp-automation/Jenkinsfile [hunk 1]
```diff
@@ -24,7 +30,7 @@ node {
     stage('Run Module Test') {
             def dockerImage = docker.image('tfp-automation')
             dockerImage.inside() {
-                sh "go test -v -timeout 1h -run ${params.TEST_CASE} ./tfp-automation/tests/cluster"
+                sh "go test -v -timeout ${timeout} -run ${env.TEST_CASE} ./tfp-automation/tests/cluster"
```
Comment (markusewalker) [inline tfp-automation/Jenkinsfile]: I think the `params.TEST_CASE` is okay to leave if you want. I am not sure if `env.TEST_CASE` will break or not.

**Review by markusewalker — APPROVED** [review]
Left one comment that is more of a comment as opposed to a concern, but else LGTM.

**Review by markusewalker — APPROVED** [review]

**Review by slickwarren — APPROVED** [review]
lgtm

**Comment by a-blender** [comment]
@Josh-Diamond Is there an issue to link this to?

**Review by a-blender — APPROVED** [review]

**Comment by markusewalker** [comment]
@a-blender yeah, it's this [issue](https://github.com/rancher/qa-tasks/issues/1056).

**Pull request merged**